flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/declarative/CancelingTest.java

```diff
@@ -0,0 +1,171 @@
+            // register execution at EG
+            ExecutingTest.MockExecutionJobVertex ejv =
+                    new ExecutingTest.MockExecutionJobVertex(canceling.getExecutionGraph());
+            TaskExecutionStateTransition update =
+                    new TaskExecutionStateTransition(
+                            new TaskExecutionState(
+                                    canceling.getJob().getJobID(),
+                                    ejv.getMockExecutionVertex()
+                                            .getCurrentExecutionAttempt()
+                                            .getAttemptId(),
+                                    ExecutionState.FAILED,
+                                    new RuntimeException()));
+            canceling.updateTaskExecutionState(update);
+            ctx.assertNoStateTransition();
+        }
+    }
+
+    private Canceling createCancelingState(MockStateWithExecutionGraphContext ctx)
+            throws JobException, JobExecutionException {
+        return createCancelingState(ctx, TestingExecutionGraphBuilder.newBuilder().build());
+    }
+
+    private Canceling createCancelingState(
+            MockStateWithExecutionGraphContext ctx, ExecutionGraph executionGraph) {
+        final ExecutionGraphHandler executionGraphHandler =
+                new ExecutionGraphHandler(
+                        executionGraph,
+                        log,
+                        ctx.getMainThreadExecutor(),
+                        ctx.getMainThreadExecutor());
+        final OperatorCoordinatorHandler operatorCoordinatorHandler =
+                new OperatorCoordinatorHandler(
+                        executionGraph,
+                        (throwable) -> {
+                            throw new RuntimeException("Error in test", throwable);
+                        });
+        executionGraph.transitionToRunning();
+        Canceling canceling =
+                new Canceling(
+                        ctx,
+                        executionGraph,
+                        executionGraphHandler,
+                        operatorCoordinatorHandler,
+                        log);
+        executionGraph.setInternalTaskFailuresListener(new TestInternalFailuresListener(canceling));
+        return canceling;
+    }
+
+    private static class TestInternalFailuresListener implements InternalFailuresListener {
+
+        private final Canceling canceling;
+
+        public TestInternalFailuresListener(Canceling canceling) {
+            this.canceling = canceling;
+        }
+
+        @Override
+        public void notifyTaskFailure(
+                ExecutionAttemptID attemptId,
+                Throwable t,
+                boolean cancelTask,
+                boolean releasePartitions) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public void notifyGlobalFailure(Throwable t) {
+            canceling.handleGlobalFailure(t);
+        }
```

Comment:
For which test do we need this implementation?

Reply:
For the `testTaskFailuresAreIgnored()` test.

I'm actually proposing to remove that test. It doesn't belong here. We need to test the proper handling of the `updateTaskExecutionState()` on the DeclarativeScheduler itself (which registers an InternalFailuresListener and forwards failures to the `handleGlobalFailure()` method of the State).
Since we are testing the `handleGlobalFailure()` in a unit test, having a test for `updateTaskExecutionState()` here doesn't make much sense.

@zentol: Can we add such a test to the DeclarativeScheduler skeleton PR?

Reply:
If I am not mistaken, then we don't need an `InternalFailuresListener` for testing the `updateTaskExecutionState`. I thought that his listener is only used when a failure in some other operation (e.g. `deploy` occurs). Testing that `updateTaskExecutionState` stays in the `CANCELING` state makes sense to me.